### PR TITLE
Implement waitTillLastBlockProcessed helper function

### PR DESCRIPTION
**Pull request by prushin**

### CODE REVIEW

File: btc-lsp/test/RpcSpec.hs [hunk 1]
```diff
@@ -21,28 +21,27 @@ spec = do
...
-        _ <- liftIO $ generateToAddress client 3 addr Nothing
-
-        --Sleep ten seconds to electrs synchronize with blockchain
-        sleep $ MicroSecondsDelay 10000000
-        elecBal <- withElectrs Rpc.getBalance ($ Left $ OnChainAddress addr)
-        liftIO $ elecBal `shouldSatisfy` isRight
-        case elecBal of
-          Right bal -> liftIO $ confirmed bal `shouldSatisfy` (> 0)
-          Left _ -> error "Error getting balance"
+  focus $
```
Comment (tim2CF) [inline btc-lsp/test/RpcSpec.hs:24]: Unfocus

**Review by tim2CF — CHANGES_REQUESTED** [review]

File: btc-lsp/test/RpcSpec.hs [hunk 2]
```diff
@@ -21,28 +21,27 @@ spec = do
...
+                    Lnd.account = Nothing
+                  }
+            )
+      case addrResponse of
+        Left _ -> error "Error getting address from lnd"
+        Right response -> do
+          elecBal <- runExceptT $ do
+            let addr = coerce response
+            _ <- withBtcT generateToAddress (\h -> h 3 addr Nothing)
+            () <- waitTillLastBlockProcessedT 100
```
[the remark this answers is not in the document]
Comment (prushin) [reply btc-lsp/test/RpcSpec.hs]: okay

File: btc-lsp/test/RpcSpec.hs (hunk 1 above)
Comment (prushin) [reply btc-lsp/test/RpcSpec.hs]: done

File: btc-lsp/src/BtcLsp/Rpc/Helper.hs [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+waitTillLastBlockProcessedT :: Env m => Integer -> ExceptT Failure m ()
+waitTillLastBlockProcessedT 0 = throwE $ FailureElectrs CannotSyncBlockchain
+waitTillLastBlockProcessedT decr = do
+  sleep $ MicroSecondsDelay 1000000
+  bHeight <- withBtcT getBlockCount id
+  bHeader <- withElectrsT Rpc.blockHeader ($ Rpc.BlockHeight (coerce bHeight))
+  case bHeader of
+    Rpc.BlockHeader "" -> waitTillLastBlockProcessedT (decr -1)
+    _ -> return ()
```
[the remark this answers is not in the document]
Comment (prushin) [reply btc-lsp/src/BtcLsp/Rpc/Helper.hs]: good idea, actually if height the same there couldn't be other block
but better double check

[the remark this answers is not in the document]
Comment (prushin) [reply btc-lsp/src/BtcLsp/Rpc/Helper.hs]: after series of investigations looks like there is the same hash, but in some other format
so doesn't match

**Review by tim2CF — APPROVED** [review]
Good!

**Pull request merged**